src/lib/djerba/plugins/fusion/plugin.py

```diff
@@ -63,12 +67,129 @@ def sort_by_actionable_level(row):
+        output_tsv_path = os.path.join(output_dir, 'fusion_blurb_urls.tsv')
+        with open(output_tsv_path, 'w', newline='') as tsvfile:
+            writer = csv.writer(tsvfile, delimiter='\t')
+            writer.writerow(['Fusion', 'Whizbam URL'])
+            writer.writerows(fusion_url_pairs)
+
         data = self.get_starting_plugin_data(wrapper, self.PLUGIN_VERSION)
         data[core_constants.RESULTS] = results
         data[core_constants.MERGE_INPUTS]['gene_information_merger'] = gene_info
         data[core_constants.MERGE_INPUTS]['treatment_options_merger'] = treatment_opts
         return data
 
+    def process_fusion(self, config, fusion, tsv_file_path, json_template_path, output_dir):
+        base_dir = (directory_finder(self.log_level, self.log_path).get_base_dir())
+        wrapper = self.get_config_wrapper(config)
+        match = re.match(r"(.+)::(.+)", fusion)
+        if match:
+            gene1 = match.group(1)
+            gene2 = match.group(2)
+        else:
+            raise ValueError(f"No valid fusion found for {fusion}.")
+
+        # Initialize breakpoints
+        breakpoint1, breakpoint2 = None, None
+
+        # Open and read the ARRIBA TSV file
+        with open(tsv_file_path, mode='r') as file:
+            reader = csv.DictReader(file, delimiter='\t')
+
+            # Find the correct row based on gene1 and gene2
+            for row in reader:
+                if (row['#gene1'] == gene1 or row['#gene1'] == gene2) and (
+                        row['gene2'] == gene1 or row['gene2'] == gene2):
+                    breakpoint1 = row['breakpoint1']
+                    breakpoint2 = row['breakpoint2']
+                    break
+
+        if not (breakpoint1 and breakpoint2):
+            raise ValueError(f"No matching fusion found in the TSV file for {fusion}.")
+
+        # Function to modify the breakpoint format to "chr:start-end"
+        def format_breakpoint(breakpoint):
+            chrom, pos = breakpoint.split(':')
+            start = int(pos)
+            end = start + 1
+            return f"{chrom}:{start}-{end}"
+
+        formatted_breakpoint1 = format_breakpoint(breakpoint1)
+        formatted_breakpoint2 = format_breakpoint(breakpoint2)
+
+        # Load the JSON template
+        with open(json_template_path, 'r') as json_file:
+            data = json.load(json_file)
+
+        # Update the JSON with the formatted breakpoints
+        data['locus'] = [formatted_breakpoint1, formatted_breakpoint2]
+
+        tumour_id = wrapper.get_my_string(core_constants.TUMOUR_ID)
+        data['tracks'][1]['name'] = tumour_id
+
+        # Search for the BAM and BAI files using glob.glob
+        bam_pattern = f"/.mounts/labs/prod/whizbam/*/RNASEQ/{tumour_id}.bam"
+        bai_pattern = f"/.mounts/labs/prod/whizbam/*/RNASEQ/{tumour_id}.bai"
+
+        bam_files = glob.glob(bam_pattern)
+        bai_files = glob.glob(bai_pattern)
+
+        if bam_files:
+            bam_file = bam_files[0]
```

Comment:
If there is more than one BAM file, is it safe to pick one at random? (If I recall correctly, `glob` doesn't guarantee order in its results.) Should we at least raise a warning if multiple BAM files are found?

Reply:
Excellent point! 
I added a warning at an earlier commit and got the message 
`Multiple BAM files found for pattern: /.mounts/labs/prod/whizbam/*/RNASEQ/LBR-0257_LCM.bam. Using the first one.`
And when I printed out the matched patterns: 
```
List of BAM files:
/.mounts/labs/prod/whizbam/BTCWGTS/RNASEQ/LBR-0257_LCM.bam
/.mounts/labs/prod/whizbam/whizbam_test/RNASEQ/LBR-0257_LCM.bam
```
We're now incorporating the project ID into the pattern, so we're not using wildcards anymore
This should ensure that the pattern matches only one file, since under whizbam dir we use symlinks for the actual files and you can't have more than one symlink with the same name (I can put the warning back if you like)

Reply:
I don't understand what "incorporating the project ID into the pattern" means? The pattern in the code is: `"/.mounts/labs/prod/whizbam/*/RNASEQ/{tumour_id}.bam"` (and similar for bai)

It is possible for this to match `"/.mounts/labs/prod/whizbam/foo/RNASEQ/{tumour_id}.bam"` and `"/.mounts/labs/prod/whizbam/bar/RNASEQ/{tumour_id}.bam"`

Saying this _shouldn't_ happen is quite different from saying it _won't ever_ happen :smile: 

As a general principle, if we expect inputs to be in a certain format, it is best to explicitly check if they are in that format, instead of assuming conventions about file naming will always be followed. Then if something breaks the convention, we will know about it, instead of having to debug mysterious errors (or worse still, not noticing the error at all).

It's easy to check that we do in fact have exactly one BAM file and one BAI file as expected, so I think we should do that.

Reply:
The pattern in the code has been changed to 
```
# Search for the BAM and BAI files using glob.glob
        bam_pattern = f"{core_constants.WHIZBAM_PATTERN_ROOT}/{project_id}/RNASEQ/{tumour_id}.bam"
        bai_pattern = f"{core_constants.WHIZBAM_PATTERN_ROOT}/{project_id}/RNASEQ/{tumour_id}.bai"
```

Reply:
Constructing an exact file path is a good solution.

This means we no longer need `glob` -- what we have here is an exact path, not a pattern. The existing code will work but is a little misleading since `glob` is normally used with wildcards -- more usual in this situation would be methods in `os.path` to check the file exists. Also see Djerba's `path_validator` class: https://github.com/oicr-gsi/djerba/blob/cdaea31c8f069f36b4b4a45b059a7bc899a8d3a9/src/lib/djerba/util/validator.py#L129

I don't think it's essential to rewrite without glob, but you might want to put a comment like `Checking existence of BAM/BAI file paths` to make it more clear what we are doing.